### PR TITLE
ENH: Adds command-line support for `importable_formats` feature in QIIME2

### LINKED ISSUE #139: Show importable formats
Related to https://github.com/qiime2/qiime2/pull/267

### PR DESCRIPTION
**Pull request by maxvonhippel**
This PR adds a feature to `q2cli` where the user can, during import, easily see a list of the available importable source formats given the plugins currently installed in their instance of `qiime2`.  This is the result of *Pear-Powered Peer-Pair Power-Trip Protocol Preventing Poor Programming Practices*-coding with @Kleptobismol and is contingent on the merging of our other pair-programmed PR, [#267 on Qiime2](https://github.com/qiime2/qiime2/pull/267).

The feature works as follows:

````BASH
$ qiime tools import --show-importable-formats

AlignedDNAFASTAFormat
AlignedDNASequencesDirectoryFormat
AlphaDiversityDirectoryFormat
AlphaDiversityFormat
BIOMV100DirFmt
BIOMV100Format
... etc 
````

We added a test immediately below (and in the style of) the test for `--show-importable-types`.  Our test checks that the values which we expect to occur with only the dummy plugin installed do occur, and that those which should not make it through indeed do not.  This is quasi-redundant, since we do similar testing in the above-linked PR for `qiime2`, but we think it makes sense given the precedent given stylistically by the implementation and testing of the similar `--show-importable-types` feature.

This PR should not be merged until the above-linked PR in `qiime2` is merged. 

Fixes #139 .

### CODE REVIEW

**Comment by maxvonhippel** [comment]
(Note that the tests will fail until the `qiime2` PR is accepted.  I can make a trivial change to a comment and push once that happens if it's desirable to have GitHub's Travis instance reflect the passing of the tests, but regardless, we've tested on my machine and @Kleptobismol and everything appears to work perfectly provided you use the version of `qiime2` from the other PR linked to and repeatedly mentioned above.)

**Comment by maxvonhippel** [comment]
This is ready for review.  If someone with write access could click the little "redo" button on the Travis check that would be awesome!  It should pass now that my changes to `qiime2` were merged.  Thanks!

File: q2cli/tests/test_cli.py [hunk 1]
```diff
@@ -77,6 +77,18 @@ def test_show_importable_types(self):
         self.assertTrue('Kennel[Dog]' in result.output)
         self.assertTrue('Mapping' in result.output)
 
+    def test_show_importable_formats(self):
+        result = self.runner.invoke(
+            tools, ['import', '--show-importable-formats'])
+        self.assertEqual(result.exit_code, 0)
+        self.assertTrue('FourIntsDirectoryFormat' in result.output)
+        self.assertTrue('IntSequenceDirectoryFormat' in result.output)
+        self.assertFalse('DummyFormat' in result.output)
```
Comment (jairideout) [inline q2cli/tests/test_cli.py]: The new format names are `UnimportableFormat` and `UnimportableDirectoryFormat`, can you assert that both of those aren't in the output?

File: q2cli/tools.py [hunk 2]
```diff
@@ -80,6 +98,9 @@ def show_importable_types(ctx, param, value):
               callback=show_importable_types, expose_value=False,
               help='Show the semantic types that can be supplied to --type '
                    'to import data into an artifact.')
+@click.option('--show-importable-formats', is_flag=True, is_eager=True,
+              callback=show_importable_formats, expose_value=False,
+              help='Show all possible source formats available for import.')
```
Comment (jairideout) [inline q2cli/tools.py]: Can you reference the `--source-format` parameter in this help text, that way users know what these formats can be plugged into? Something like: *Show formats that can be supplied to --source-format to import data into an artifact.*

File: q2cli/tools.py [hunk 3]
```diff
@@ -55,6 +55,24 @@ def show_importable_types(ctx, param, value):
     ctx.exit()
 
 
+def show_importable_formats(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+
+    import qiime2.sdk
+
+    importable_formats = list(qiime2.sdk.PluginManager().importable_formats)
```
Comment (jairideout) [inline q2cli/tools.py]: Can you display the formats in sorted order?

```python
importable_formats = sorted(qiime2.sdk.PluginManager().importable_formats)
```

**Comment by jairideout** [comment]
@maxvonhippel I reran the Travis-CI tests and everything's green!

File: q2cli/tools.py (hunk 2 above)
Comment (maxvonhippel) [inline q2cli/tools.py]: Good point, will do.

**Comment by maxvonhippel** [comment]
Thanks @jairideout !  I'll do all this this afternoon and push.

File: q2cli/tools.py (hunk 2 above)
Comment (maxvonhippel) [inline q2cli/tools.py]: Gotcha, yeah that's better as it matches the precedent style from `importable types`.  

And done :D

**Comment by maxvonhippel** [comment]
Ready for review!

**Pull request merged**